agent/proxycfg/snapshot.go

```diff
@@ -641,6 +640,55 @@ func (c *configSnapshotMeshGateway) isEmptyPeering() bool {
 		!c.PeeringTrustBundlesSet
 }
 
+type upstreamIDSet map[UpstreamID]struct{}
+
+func (u upstreamIDSet) add(uid UpstreamID) {
+	u[uid] = struct{}{}
+}
+
+type routeUpstreamSet map[structs.ResourceReference]upstreamIDSet
+
+func (r routeUpstreamSet) has(uid UpstreamID) bool {
```

Comment:
I would rename this since `set.has()` reads as though it's checking for a key in a map instead of looping through all the values
```suggestion
func (r routeUpstreamSet) hasUpstream(uid UpstreamID) bool {
```